server/routes/logic.js

```diff
@@ -0,0 +1,123 @@
+require('dotenv').config();
+const bcrypt = require('bcrypt');
+const saltRounds = 10;
+const db = require('../db/db'); // ../db/db is based on how far from index.js
+const jwt = require('jsonwebtoken');
+const cookie = require("cookie");
+
+// Check if the user is already authenticated
+const routeAccessibility = async (req, res, next) => {
+  const authCookie = req.cookies.AccessToken;
+  if (authCookie) {
+    jwt.verify(authCookie, process.env.ACCESS_TOKEN_SECRET, (err, user) => { // user is a value serialised wich is authenticatedUser
+      if (err) return res.sendStatus(403);
+      res.redirect('/authenticated');
+    });
+  } else {  
+    next();
+  }
+};
+
+// Check user existense
+const signIn =  async (req, res) => {
+  if (req.body.user_name) {
+    const name = req.body.user_name;
+    const pw = req.body.password;
+    const data = await db.checkIn(name);
+    if (data) {
+      const id = data.user_id
+      const uName = data.user_name
+      const err = data.err;
+      if (id && uName) {
+        await bcrypt.compare(pw, data.password).then(async (result) => {
+        const authenticatedUser = { id: id, name: uName };
+        if (result) {
+          const accessToken = await setToken(authenticatedUser);
+          res.setHeader("Set-Cookie", cookie.serialize("AccessToken", accessToken,  { httpOnly: true }))
```

Comment:
Does this set the header as `Set-Cookie` or `AccessToken`  I'm a little unsure 🤔

Reply:
`Set-Cookie` is the key, and the `cookie.serialize("AccessToken", accessToken,  { httpOnly: true })` is the value

Reply:
This is the combination of the 
[res.setHeader](https://nodejs.org/docs/v0.4.0/api/http.html#response.setHeader) and [Set-Cookie](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie)

Reply:
I might not getting your question straight so let me know if I'm not answering to your point.